content/blog/2024-06-18-gsoc-lxboot_x86.mdx

```diff
@@ -0,0 +1,41 @@
+---
+title: "Linux x86 Boot Protocol Support"
+description: |
+  The Linux boot protocol plays an important role in the initialization of the Linux operating system, emphasizing the importance of system optimization and scalability.
+publishedDate: 2024-06-18
+tags:
+- gsoc
+- gsoc24
+- Linux x86
+- Booting
+authors:
+- Mihnea Firoiu
+---
+
+<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-generic-badge.svg" align="right" />
+
+## GSoC'24: Linux x86 Boot Protocol Support
+
+### Project overview
+
+The Linux boot protocol plays an important role in the initialization of the Linux operating system, emphasizing the importance of system optimization and scalability.
+This system ensures a seamless transfer of control from the bootloader to the Linux kernel by providing essential information including memory map, command line parameters, and device specifications.
+By adhering to a uniform boot protocol, Linux achieves a level of versatility across the hardware spectrum of architectures and bootloader functions.
+Essentially, this protocol is a commitment to a standardized approach that is not only stable but also facilitates bootloader development, improving compatibility and reliability.
+In particular, the Linux boot protocol greatly contributes to the overall robustness and reliability of the bootstrapping process on Linux-based systems. Almost, if not all, widely
+ available bootloaders that exist on the market are able to speak this protocol and Unikraft, being unable to be detected by these bootloaders as a Linux Boot Protocol compliant system,
+ is hindering the ability of Unikraft to boot on a very large amount of platforms.
+
+### Current Progress
+
+I already compiled `qemu 9.0.1` using the following flags: `--enable-debug --extra-cflags="-g3" --extra-ldflags="-g3" --disable-strip --enable-kvm --enable-avx2 --enable-avx512bw --enable-avx512f` to help me debug.
+At the moment, I am looking into creating `support/scripts/mklinux_x86.py` using as inspiration the already existing `mklinux.py` and the [ARM64 PR](https://github.com/unikraft/unikraft/pull/988).
+
+### Next Steps
+
+Once everything works for QEMU, I will move onto GRUB2 and look into making lxboot x86 work with it as ISO and as Disk.
```

Comment:
Try to add a few more detailed steps on what needs to happen in order for this to run on `qemu`. What the `mklinux_x86.py` script will contain, what the header of the image will look like, etc.